sdcm/cluster_aws.py

```diff
@@ -340,7 +340,8 @@ def update_bootstrap(ec2_user_data, enable_auto_bootstrap):
         return ec2_user_data
 
     def _create_or_find_instances(self, count, ec2_user_data, dc_idx):
-        if self.nodes:
+        region_names = self.params.get('region_name')
+        if [node for node in self.nodes if node.region == region_names[dc_idx]]:
```

Comment:
I even remember this discussion, it was agreed that we are doing it this way, to not complicate the PR.
Your solution would no work, since when SCT is just starting there are no such nodes in the given DC, and you fall into `self._create_instances(count, ec2_user_data, dc_idx)`.

I think code need to address all the cases - i.e. partially provisioned instances ... etc..
In order to do that you need to have something like this:
```
    def _create_or_find_instances(self, count, ec2_user_data, dc_idx):
        already_in_use = [node._instance.instance_id for node in self.nodes if node.dc_idx == dc_idx]
        unused_on_cloud = [i for i in self._get_instances(dc_idx) if i.instance_id in already_in_use]
        if len(unused_on_cloud) <= count:
            self.log.info('Found provisioned instances = %s', unused_on_cloud[:count])
            return unused_on_cloud[:count]

        newly_provisioned = self._create_instances(count - len(unused_on_cloud), ec2_user_data, dc_idx)
        self.log.info('Found provisioned instances = %s, and provisioned additional instances = %s ', unused_on_cloud, newly_provisioned)
        return unused_on_cloud + newly_provisioned
```